homeassistant/components/cover/velux.py

```diff
@@ -0,0 +1,96 @@
+"""
+Support for Velux covers.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/cover.velux/
+"""
+
+from homeassistant.components.cover import (
+    ATTR_POSITION, SUPPORT_CLOSE, SUPPORT_OPEN, SUPPORT_SET_POSITION,
+    SUPPORT_STOP, CoverDevice)
+from homeassistant.components.velux import DATA_VELUX
+from homeassistant.core import callback
+
+DEPENDENCIES = ['velux']
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up cover(s) for Velux platform."""
+    entities = []
+    for node in hass.data[DATA_VELUX].pyvlx.nodes:
+        from pyvlx import OpeningDevice
+        if isinstance(node, OpeningDevice):
+            entities.append(VeluxCover(node))
+    async_add_entities(entities)
+
+
+class VeluxCover(CoverDevice):
+    """Representation of a Velux cover."""
+
+    def __init__(self, node):
+        """Initialize the cover."""
+        self.node = node
+
+    @callback
+    def async_register_callbacks(self):
+        """Register callbacks to update hass after device was changed."""
+        async def after_update_callback(device):
```

Comment:
Question, should the `async def` be outside of the `def`? I'm not 100% familiar with this yet.
example: [cover.template.pyL201](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/cover/template.py#L201)

Reply:
I think that's okay, as it is just passed as a callback (which I assume gets awaited later on). The @callback marks this as something that does not do I/O (for optimization), see: https://developers.home-assistant.io/docs/en/asyncio_categorizing_functions.html#the-callback-function